torch/ao/quantization/fx/_model_report/model_report.py

```diff
@@ -7,22 +7,78 @@
 
 class ModelReport:
     r"""
-    Generates report and collects statistics
-        Used to provide users suggestions on possible model configuration improvements
+    The ModelReport class aims to provide users an easy way to diagnose issues that they run into
+    with their models. The class works with all traceable GraphModules to help diagnose issues,
+    though the requirements on the type of model more-so depends on the specific report the user
+    is trying to generate. With respect to the reports, the ModelReport class is intialized with
+    a set of Detector classes, each of which generate reports on quantization configuration
+    issues a use might have.
 
     Currently supports generating reports on:
+    - Suggestions for per-channel vs. per-tensor quantization (nn.Module)
     - Suggestions for dynamic vs static quantization for linear layers (Graph Modules)
+    - Suggestions for input-weight equalization for linear and conv layers (Graph Modules)
+    - Suggestions for outlier detection for all layers (Graph Modules)
```

Comment:
are these supported by `ModelReport` or detectors?

Reply:
These are technically by the detectors, but since the detectors are used through the ModelReport API, I thought this would still count as support by the ModelReport API.

Basically, I want the user to primarily (and hopefully exclusively) be interacting with the ModelReport class and not the Detectors